packages/prisma/schema.prisma

```diff
@@ -238,3 +241,51 @@ model DocumentShareLink {
 
   @@unique([documentId, email])
 }
+
+enum TemplateType {
+  PUBLIC
+  PRIVATE
+}
+
+model Template {
+  id                     Int                 @id @default(autoincrement())
+  userId                 Int
+  User                   User                @relation(fields: [userId], references: [id], onDelete: Cascade)
+  title                  String
+  type                   TemplateType        @default(PRIVATE)
+  TemplateRecipient      TemplateRecipient[]
+  TemplateField          TemplateField[]
```

Comment:
We can probably reuse the existing models for this rather than creating new ones. Doing so means we don't need to keep them both up to date as we change things around.

Realistically a template is just a special document and the rest of our logic is in code 😄

Reply:
The existing Recipient had extra things which I did not need in the templates that made it difficult for me to work with so I created this model for the things I needed. The Recipient model has an expired, signedAt, readStatus, signingStatus, sendStatus which are required but then I don't need them for the templates

For the Field, the documentId was the tie-breaker. I didn't want to remove it and I did not have it in some cases but it is possible because the Field and TemplateField are the same, the only difference is the documentId and templateId so I think I can work with it.

Reply:
For recipients I'd just null them out in the code. 

For the other items we can probably make the relation nullish and have a templateId with the same nullish relation.

If you create the migration with a `--create-only` flag we can also try using a CHECK constraint to ensure either documentId or templateId exist during creation/updates